packages/hash/api/src/graph/workspace-types.ts

```diff
@@ -0,0 +1,136 @@
+import {
+  GraphApi,
+  EntityType,
+  PropertyType,
+} from "@hashintel/hash-graph-client";
+import { Logger } from "@hashintel/hash-backend-utils/logger";
+import { AxiosError } from "axios";
+import { DataTypeModel, EntityTypeModel, PropertyTypeModel } from "../model";
+import { logger } from "../logger";
+import {
+  primitiveDataTypeVersionedUris,
+  generateSchemaBaseUri,
+  generateWorkspacePropertyTypeSchema,
+  generateWorkspaceEntityTypeSchema,
+  worskspaceTypesNamespaceUri,
+  workspaceAccountId,
+} from "../model/util";
+
+// Generate the schema for the shortname property type
+const shortnamePropertyType = generateWorkspacePropertyTypeSchema({
+  title: "Shortname",
+  possibleValues: [{ primitiveDataType: "Text" }],
+});
+
+// Generate the schema for the email property type
+const emailPropertyType = generateWorkspacePropertyTypeSchema({
+  title: "Email",
+  possibleValues: [{ primitiveDataType: "Text" }],
+});
+
+const workspacePropertyTypes: PropertyType[] = [
+  shortnamePropertyType,
+  emailPropertyType,
+];
+
+// Generate the schema for the user entity type
+const userEntityType = generateWorkspaceEntityTypeSchema({
+  title: "User",
+  properties: [
+    {
+      baseUri: generateSchemaBaseUri({
+        namespaceUri: worskspaceTypesNamespaceUri,
+        kind: "propertyType",
+        title: "Shortname",
+      }),
+      versionedUri: shortnamePropertyType.$id,
+    },
+    {
+      baseUri: generateSchemaBaseUri({
+        namespaceUri: worskspaceTypesNamespaceUri,
+        kind: "propertyType",
+        title: "Email",
+      }),
+      versionedUri: emailPropertyType.$id,
+      required: true,
+      /** @todo: un-comment the next two lines when a related Graph API bug is fixed */
+      //   array: true,
+      //   minItems: 1,
```

Comment:
Fixed in https://github.com/hashintel/hash/pull/917, thanks @Alfred-Mountfield!

Will uncomment now